.github/workflows/server.yml

```diff
@@ -0,0 +1,133 @@
+name: Server
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - 'server/**'
+      - '.github/workflows/server.yml'
+  pull_request:
+    paths:
+      - 'server/**'
+      - '.github/workflows/server.yml'
+  workflow_dispatch:
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        description: "Version for the build, without v prefix"
+        required: false
+
+env:
+  VERSION: ${{ inputs.version }}
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./server
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Installing dependencies
+        uses: ./.github/actions/install
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.59
+          args: --timeout=3m
+          working-directory: ./server
+
+      - name: Install gofumpt
+        run: go install mvdan.cc/gofumpt@latest
```

Comment:
prefer set a specific version instead of latest.
It will avoid the things breaking without any change in basecode